src/main/java/com/semmle/jira/addon/LgtmServlet.java

```diff
@@ -144,46 +177,34 @@ void createIssue(Request request, HttpServletResponse resp, ProcessedConfig conf
     } else {
       IssueService.IssueResult issueResult =
           ComponentAccessor.getIssueService().create(config.getUser(), createValidationResult);
-      LabelManager mgr = ComponentAccessor.getComponent(LabelManager.class);
-      mgr.addLabel(config.getUser(), issueResult.getIssue().getId(), "LGTM", false);
-      mgr.addLabel(config.getUser(), issueResult.getIssue().getId(), request.project.name, false);
 
       Response response = new Response(issueResult.getIssue().getId());
       sendJSON(resp, HttpServletResponse.SC_CREATED, response);
     }
   }
 
   void applyTransition(
-      Long issueId, HttpServletResponse resp, Status targetStatus, ProcessedConfig config)
+      MutableIssue issue, String transitionName, HttpServletResponse resp, ProcessedConfig config)
       throws IOException {
-
-    MutableIssue issue =
-        ComponentAccessor.getIssueService().getIssue(config.getUser(), issueId).getIssue();
-
-    if (issue == null) {
-      sendError(resp, HttpServletResponse.SC_NOT_FOUND, "Issue " + issueId + " not found.");
-      return;
-    }
-
-    if (issue.getStatus().getId() == targetStatus.getId()) {
-      Response response = new Response(issue.getId());
-      sendJSON(resp, HttpServletResponse.SC_OK, response);
-      return;
-    }
-
-    JiraWorkflow workflow = ComponentAccessor.getWorkflowManager().getWorkflow(issue);
-
-    Collection<ActionDescriptor> candidateActions =
-        workflow.getActionsWithResult(workflow.getLinkedStep(targetStatus));
-
     IssueInputParameters issueInputParameters =
         ComponentAccessor.getIssueService().newIssueInputParameters();
     issueInputParameters.setRetainExistingValuesWhenParameterNotProvided(true, true);
 
+    JiraWorkflow workflow = ComponentAccessor.getWorkflowManager().getWorkflow(issue);
+    Collection<ActionDescriptor> candidateActions = workflow.getActionsByName(transitionName);
```

Comment:
Is this still necessary now we have/expect control of the workflow? Could we just do something like...

```ActionDescriptor action = Iterables.getOnlyElement(workflow.getActionsByName(transitionName));```

Reply:
Yes, we could do that, but, users will always be able to rename/delete the transition or add a new one with the same name, and I'm don't believe `Iterables.getOnlyElement` will work correctly in all those cases.

Reply:
I guess it's a question of whether we _want_ things to fail if the user starts messing about with the workflow. Ideally they won't touch it, but I imagine someone will...